File: src/main/java/requests/spark/websockets/objects/messages/outgoing/QuestionResults.java
package requests.spark.websockets.objects.messages.outgoing;

import game.actors.Question;
import game.actors.QuestionOption;
import org.apache.log4j.Logger;
import org.json.JSONArray;
import org.json.JSONObject;
import requests.spark.websockets.objects.Message;
import requests.spark.websockets.objects.MessageType;

@MessageType("QuestionResults")
public class QuestionResults extends Message {
    private static Logger log = Logger.getLogger(QuestionResults.class);

    private Question question = null;

    public QuestionResults question(Question question) {
        this.question = question;
        return this;
    }

    public void prepareToSend() {
        addResponseData("questionUUID", question.getUuid());

        JSONArray options = new JSONArray();
        for (QuestionOption questionOption : question.getPossibleOptions()) {
            JSONObject option = new JSONObject();

            option.put("optionUUID", questionOption.getUuid());
            option.put("optionAnswerCount", questionOption.getCountedAnswers());
            options.put(option);
        }

        addResponseData("options", options);
        addResponseData("correctOptionUUID", question.getCorrectOption().getUuid());
    }
}

File: src/main/ngapp/src/app/wsObjects/questionResults.ts
import {Message} from "./message";
import {QuestionOption} from "../questionOption";
import {PlayerResult} from "../playerResult";

export class QuestionResults extends Message {

  private _questionUUID: string;
  private _correctOptionUUID: string;
  private _questionOptions: Array<QuestionOption> = [];
  private _playerResults: Array<PlayerResult> = [];

  constructor() {
    super();
    this.type = "QuestionResults";
  }

  decodeResponse(msgRaw: any) {
    this._questionUUID = msgRaw.questionUUID;
    this._questionOptions = msgRaw.options;
    this._correctOptionUUID = msgRaw.correctOptionUUID;
    this._playerResults = msgRaw.playerResults;
  }

  get questionUUID(): string {
    return this._questionUUID;
  }

  get correctOptionUUID(): string {
    return this._correctOptionUUID;
  }

  get questionOptions(): Array<QuestionOption> {
    return this._questionOptions;
  }

  get playerResults(): Array<PlayerResult> {
    return this._playerResults;
  }
}
